src/artemis/external_interaction/nexus/nexus_utils.py

```diff
@@ -0,0 +1,170 @@
+"""
+Define beamline parameters for I03, Eiger detector and give an example of writing a
+gridscan.
+"""
+from __future__ import annotations
+
+import time
+from datetime import datetime
+
+from dodal.devices.detector import DetectorParams
+from dodal.devices.fast_grid_scan import GridScanParams
+from nexgen.nxs_utils import Attenuator, Axis, Beam, Detector, EigerDetector, Goniometer
+
+from artemis.external_interaction.ispyb.ispyb_dataclass import IspybParams
+from artemis.parameters.plan_specific.rotation_scan_internal_params import (
+    RotationScanParams,
+)
+
+
+def get_current_time():
+    return datetime.utcfromtimestamp(time.time()).strftime(r"%Y-%m-%dT%H:%M:%SZ")
+
+
+def create_gridscan_goniometer_axes(
+    detector_params: DetectorParams, grid_scan_params: GridScanParams, grid_scan: dict
+) -> Goniometer:
+    """Create the data for the goniometer.
+
+    Args:
+        detector_params (DetectorParams): Information about the detector.
+        grid_scan_params (GridScanParams): Information about the experiment.
+        grid_scan (dict): scan midpoints from scanspec.
+
+    Returns:
+        Goniometer: A Goniometer description for nexgen
+    """
+    # Axis: name, depends, type, vector, start
+    gonio_axes = [
+        Axis("omega", ".", "rotation", (-1.0, 0.0, 0.0), detector_params.omega_start),
+        Axis(
+            "sam_z",
+            "omega",
+            "translation",
+            (0.0, 0.0, 1.0),
+            grid_scan_params.z_axis.start,
+            grid_scan_params.z_axis.step_size,
+        ),
+        Axis(
+            "sam_y",
+            "sam_z",
+            "translation",
+            (0.0, 1.0, 0.0),
+            grid_scan_params.y_axis.start,
+            grid_scan_params.y_axis.step_size,
+        ),
+        Axis(
+            "sam_x",
+            "sam_y",
+            "translation",
+            (1.0, 0.0, 0.0),
+            grid_scan_params.x_axis.start,
+            grid_scan_params.x_axis.step_size,
+        ),
+        Axis("chi", "sam_x", "rotation", (0.006, -0.0264, 0.9996), 0.0),
+        Axis("phi", "chi", "rotation", (-1, -0.0025, -0.0056), 0.0),
+    ]
+    return Goniometer(gonio_axes, grid_scan)
+
+
+def create_rotation_goniometer_axes(
```

Comment:
Should: Can we refactor the common parts out between this and the grid scan gonio function?

Reply:
there's not really much to easily factor out imo, for any given experiment type it's different fields which will have to be filled in - the common stuff is already in `Axis` etc.

Reply:
on discussion,it is possible and it will be easier to do with a scanspec added, but I think it belongs more with the nexus writing stuff so I'll do that in the following PR #697